### PR TITLE
[fix] no error for local broadcast

### LINKED ISSUE #19: primus.forward shouldn't answer with error for empty servers list
primus/metroplex adds the `broadcast` function to `primus.forward`, which just gathers all servers in the metroplex registry and calls `primus.forward`. But there is a problem when you want to boradcast and the only server available is the current one processing the `forward.broadcast` call: `metroplex.servers` will return empty list (since it ignores current server) and it will call `forward` with an empty list.

The problem is even when `forward` will do everything you would expect (passing the message to every spark on the current server) it will still [call the callback with an error saying "No servers provided"](https://github.com/primus/omega-supreme/blob/master/supreme.js#L119). So the `broadcast` will also fail, even considering the scenario is not something unusual (a cluster of current size 1, maybe because you don't need too much servers at the moment or you want a cluster setup in case you need to add more nodes).

At first I thought this should be primus/metroplex 's problem, but looking at the code there is not much you can do there… if you don't call `forward` you won't deliver the message to the sparks on the current server, and you won't have the stats you need to answer with. And if you do add the current server to the servers provided to `forward`, then for a broadcast type of message it will still call the current server even when all messages were locally delivered.

So, the only solution I can think of is to not answer with error in `primus.forward` for empty servers list. It kind of makes sense to me for the fact this function will always try to deliver to local sparks… so having empty servers list is like saying "please forward to local clients". An alternative would be to have a special parameter or another function that will allow you to only deliver to local sparks, so in metroplex you can call it instead of forward if you see there are no other servers but this one (although I think it's better to just support empty servers list because of the job done in `forward`).

Does this make sense? I can submit a PR if it does, but I would like some advice on the best solution to this unexpected error.

### PR DESCRIPTION
**Pull request by raulmt**
When no sparks are sent it is considered as broadcast. If besides that no servers
are given then it should be considered as a local broadcast (it is actually doing that currently)
and a response with `local: true` and no error should be given to the callback.

This fixes #19.

### CODE REVIEW

File: supreme.js [hunk 1]
```diff
@@ -104,13 +105,18 @@ supreme.server = function server(primus, options) {
...
     }
 
     //
     // Everything was broad casted locally, we can bail out early, which is
     // naaaais <-- say out loud with Borat's voice.
     //
-    if (type !== 'broadcast' && !sparks.length) return fn(undefined, {
+    if (
+         type !== 'broadcast' && !sparks.length // no more sparks to send this message to
+      || type === 'broadcast' && local          // only local broadcast requested
```
Comment (lpinca) [inline supreme.js]: We don't need to check again `type === 'broadcast'` no? `local` is set to true only if `type` is `'broadcast'`.

File: test.js [hunk 2]
```diff
@@ -316,6 +316,45 @@ describe('omega supreme', function () {
...
+        client.on('open', next);
+      }, function (err) {
+        if (err) return next(err);
+
+        async.each(connections, function (client, next) {
+          client.on('data', function (msg) {
+            assume(msg).to.equal('bar');
+            next();
+          });
+        }, function(err) {
```
Comment (lpinca) [inline test.js]: Missing space after `function`.

File: supreme.js [hunk 3]
```diff
@@ -104,13 +105,18 @@ supreme.server = function server(primus, options) {
         spark.write(msg);
         calls++;
       });
+      // since there are no sparks, having no servers means *local* broadcast only
```
Comment (lpinca) [inline supreme.js]: Can you please make this comment consistent with the others?
```
//
// Comment with 80 chars per line.
//
```

Comment (raulmt) [inline supreme.js]: sure, I missed this, I was trying to keep the code style ;) (btw, a linter config would be helpful)

Comment (lpinca) [inline supreme.js]: I noticed and you did a good job. Thanks.

File: supreme.js (hunk 1 above)
Comment (raulmt) [inline supreme.js]: in terms of current logic, no, we don't… the intent for the `local` variable was maybe also use it for another "local only" detection, even with no broadcast… but currently I agree I left it in a weird state.

I have two proposals about this:
- change the `local` variable to `localBroadcast` in which case we are more explicit about the intention there
- keep it as `local` but also set it in the other blocks above so we can change this if condition you are commenting on to just `if (local)`

I'll go with the 2nd one and update the PR… I can go back to 1st one if you think it's better.

File: test.js (hunk 2 above)
Comment (raulmt) [inline test.js]: :O yeah… too used to eslint telling me that xD I'll fix it

**Comment by raulmt** [comment]
Done with the changes @lpinca . Please let me know what you think, specially about the `local` variable use now.

**Comment by lpinca** [comment]
I have only 1 more OCD. Can you please start the comments with a Capital letter and end them with a dot?

**Comment by lpinca** [comment]
I'll fix the remaining nits myself :)

Thanks!

**Comment by lpinca** [comment]
Released 2.1.1 with this fix.

**Comment by raulmt** [comment]
haha, thanks!

**Pull request merged**